### ISSUE TITLE
A generic error occurred in GDI+.

### ISSUE THREAD

**Issue by hanjo** [philhansen/WallpaperChanger#6]
Hi,

today I was presented with the following error message:
```
<unexpected error>
A generic error occurred in GDI+.
```
Knowing that this isn't developed any further, I looked into the code and googling a little bit revealed that this is likely some issue with writing the temporary file the app uses. In fact, for some reason, %TEMP%\wallpaper.png could not be written. So I removed the file manually and it worked again.

The fix should probably be to remove the file before writing (or just writing to another filename), but since I cannot change the code myself, I worked around it by adding the following into my PowerShell script that is calling the app:
```PowerShell
Remove-Item -Path $env:TEMP\wallpaper.png -Force -ErrorAction SilentlyContinue
```

Hope this helps someone.

**Comment by philhansen** [comment]
Hi hanjo, what operating system did this happen on?

**Comment by hanjo** [comment]
This is Windows 11 22H2 (OS Build 22621.3447). It used to work for more than ~1 year without any issues, this happened the first time.

**Comment by philhansen** [comment]
I'm not running Windows 11 yet but I wonder if it is related to some change with that OS (like maybe a permission thing).

**Comment by hanjo** [comment]
It did work for quite some time on Windows 11 and I checked the permissions of the file before I removed it, but they were just normal (i.e. my user had full read/write permissions). Also, after I removed the file I ran the app a couple of times without removing the temp file in between and there were no issues. I cannot reproduce the problem unfortunately, I just wanted to leave a hint here that the error message `A generic error occurred in GDI+.` might be caused by not being able to write this temporary file.
